rust/datafusion/src/physical_plan/planner.rs

```diff
@@ -218,32 +217,18 @@ impl DefaultPhysicalPlanner {
                     .collect::<Result<Vec<_>>>()?;
                 let aggregates = aggr_expr
                     .iter()
-                    .map(|e| {
-                        tuple_err((
-                            self.create_aggregate_expr(e, &input_schema, ctx_state),
-                            e.name(&input_schema),
-                        ))
-                    })
+                    .map(|e| self.create_aggregate_expr(e, &input_schema, ctx_state))
                     .collect::<Result<Vec<_>>>()?;
 
-                let initial_aggr = HashAggregateExec::try_new(
+                let initial_aggr = Arc::new(HashAggregateExec::try_new(
                     AggregateMode::Partial,
                     groups.clone(),
                     aggregates.clone(),
                     input,
-                )?;
+                )?);
 
-                if initial_aggr.output_partitioning().partition_count() == 1 {
```

Comment:
This seems major for a single partition.

Reply:
I a not sure I understand what you are getting at with this comment

Reply:
Sorry. What I meant is that compared to the previous implementation, we double the amount of work, even when we did not need because there is a single partition.

IMO we should handle this in another way, e.g. via a physical optimizer that re-writes the partial aggregation by a final aggregation when the input is a single partition.

Reply:
I agree, the planner/optimizer should prevent this.